test/network/scenarios/p2p/peer_disconnect.js

```diff
@@ -116,23 +124,33 @@ module.exports = function(
 				// Need to keep one peer so that we can validate
 				// Duplicate socket connection exists or not
 				it('stop all the nodes in the network except node_0', done => {
+					const peersPromises = [];
 					for (let i = 1; i < TOTAL_PEERS; i++) {
-						common.stopNode(`node_${i}`);
+						peersPromises.push(common.stopNode(`node_${i}`));
 					}
-					setTimeout(() => {
-						console.info('Wait for nodes to be stopped');
-						done();
-					}, 10000);
+					console.info('Wait for nodes to be stopped');
+					Promise.all(peersPromises)
+						.then(() => {
+							done();
+						})
+						.catch(err => {
+							done(err.message);
+						});
```

Comment:
This would be much compact option.

```js
it('stop all the nodes in the network except node_0', () => {
    return Promise.map(new Array(TOTAL_PEERS), (value, index) => {
        return common.stopNode(`node_${index + 1}`); 
    });
});
```

Reply:
Just remove the `+1` from the index.

Reply:
But then it stops `node_0` which we don't want.